src/lib/y2security/security_policies/policy.rb

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) [2022] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2security/security_policies/validator"
+
+module Y2Security
+  module SecurityPolicies
+    # This class represents a security policy
+    #
+    # It offers an API to get the security policies and run validations.
+    #
+    # @example Get all known security policies
+    #   Policy.all #=> [#<Y2Security::Policy...>]
+    #   Policy.all.map(&:name) #=> ["Defense Information Systems Agency STIG"]
+    #
+    # @example Run DISA STIG networking validation
+    #   policy = Policy.find(:disa_stig)
+    #   policy.validate.map(&:to_message) #=> ["Wireless devices are not allowed"]
+    class Policy
+      # @return [Symbol] Security policy ID
+      attr_reader :id
+      # @return [String] Security policy name
+      attr_reader :name
+      # @return [Array<String>] Security policy packages needed
+      attr_reader :packages
+      # @return [Array<Issue>] Found security policy issues
+      attr_reader :issues
```

Comment:
NP: I am not sure whether having a `#issues` method is a good idea. You never know if the `#validate` method was already called or not.

Reply:
I would suggest to always use `#validate` to get the list of issues. And I would delegate to the caller the responsibilty of caching the issues. What do you think?

Reply:
Initially, we considered using an ActiveRecord-like API (with `#valid?` and `#issues`). Then we moved the list of issues to the caller (as you have described). While refactoring the proposal (to support multiple policies) we move them back to the policy. But I agree, I would prefer to keep the list in the caller.